library/src/main/java/com/rockerhieu/rvadapter/endless/EndlessRecyclerViewAdapter.java

```diff
@@ -17,27 +17,48 @@
  */
 public class EndlessRecyclerViewAdapter extends RecyclerViewAdapterWrapper {
     public static final int TYPE_PENDING = 999;
-    private final Context context;
-    private final int pendingViewResId;
-    private AtomicBoolean keepOnAppending;
-    private AtomicBoolean dataPending;
-    private RequestToLoadMoreListener requestToLoadMoreListener;
+    private final AtomicBoolean keepOnAppending;
+    private final AtomicBoolean dataPending;
+    private final RequestToLoadMoreListener requestToLoadMoreListener;
+    private boolean shouldNotifyAdapter = false;
+    private ViewHolder pendingViewHolder;
+    private int pendingViewId = R.layout.item_loading;
+
+    public EndlessRecyclerViewAdapter(Adapter wrapped, RequestToLoadMoreListener requestToLoadMoreListener) {
+        this(wrapped, requestToLoadMoreListener, true);
+    }
 
-    public EndlessRecyclerViewAdapter(Context context, Adapter wrapped, RequestToLoadMoreListener requestToLoadMoreListener, @LayoutRes int pendingViewResId, boolean keepOnAppending) {
+    public EndlessRecyclerViewAdapter(Adapter wrapped, RequestToLoadMoreListener requestToLoadMoreListener,
+          boolean keepOnAppending) {
         super(wrapped);
-        this.context = context;
         this.requestToLoadMoreListener = requestToLoadMoreListener;
-        this.pendingViewResId = pendingViewResId;
         this.keepOnAppending = new AtomicBoolean(keepOnAppending);
         dataPending = new AtomicBoolean(false);
     }
 
-    public EndlessRecyclerViewAdapter(Context context, Adapter wrapped, RequestToLoadMoreListener requestToLoadMoreListener) {
-        this(context, wrapped, requestToLoadMoreListener, R.layout.item_loading, true);
+    /**
+     * Set layout Id in order to inflate your custom pending view.
+     * If not set, default is R.layout.item_loading
+     */
+    public void setPendingViewId(@LayoutRes int layoutId) {
```

Comment:
Once pendingView is created, it will be ignored anyway. I wouldn't mind to keep it as setter but i can move it to constructor if you feel.